java_src/foam/dao/FileJournal.java

```diff
@@ -0,0 +1,102 @@
+package foam.dao;
+
+import foam.core.Detachable;
+import foam.core.FObject;
+import foam.core.Journal;
+import foam.lib.json.JournalParser;
+import foam.lib.json.Outputter;
+
+import java.io.*;
+
+/**
+ * Created by carlos on 2017-05-22.
+ */
+public class FileJournal implements Journal {
+
+    protected BufferedWriter bw;
+    protected BufferedReader br;
+    protected File file;
+
+    public FileJournal(String filename) throws IOException {
+
+        this.file = new File(filename);
+
+        if (!file.exists()) {
+            file.createNewFile();
+        }
+
+        this.bw = new BufferedWriter(new FileWriter(file, true));
+        this.br = new BufferedReader(new FileReader(file));
+    }
+
+    /**
+     * Persists data into the File System.
+     *
+     * @param obj
+     * @param sub
+     */
+    @Override
+    public void put(FObject obj, Detachable sub) {
+        try {
+            Outputter outputter = new Outputter();
+            this.bw.write("p(foam.json.parse(" + outputter.stringify(obj) + "))");
+            this.bw.newLine();
+            this.bw.flush();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * deletes the entire journal file
+     */
+    public void removeAll() {
+        this.file.delete();
+    }
+
+    @Override
+    public void remove(FObject obj, Detachable sub) {
+        try {
+            Outputter outputter = new Outputter();
+            this.bw.write("r(foam.json.parse(" + outputter.stringify(obj) + "))");
```

Comment:
I don't see how the entire object would be reconstructed if we only save the `id` in the journal file, for example when the journal file already exist and it's being replayed back to the delegated DAO. Am I missing something ?